File: app/src/main/java/com/example/tradeup/data/repository/RatingRepository.java
package com.example.tradeup.data.repository;

import androidx.annotation.NonNull;
import com.example.tradeup.core.utils.Callback;
import com.example.tradeup.data.model.Rating; // Model Rating (Java)
import java.util.List;

public interface RatingRepository {
    void submitRating(@NonNull Rating rating, Callback<Void> callback);

    void getRatingsForUser(String userId, long limit, Callback<List<Rating>> callback);

    void getRatingForTransaction(String transactionId, String raterId, Callback<Rating> callback); // Rating có thể null
}

File: app/src/main/java/com/example/tradeup/data/repository/RatingRepository.kt
package com.example.tradeup.data.repository

import com.example.tradeup.data.model.Rating // Giả sử bạn có model Rating.kt

interface RatingRepository {
    suspend fun submitRating(rating: Rating): Result<Unit>
    suspend fun getRatingsForUser(userId: String): Result<List<Rating>>
    suspend fun getRatingForTransaction(transactionId: String, raterId: String): Result<Rating?>
}